inc/site-health.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Pantheon Site Health Modifications
+ *
+ * @package pantheon
+ */
+
+namespace Pantheon\Site_Health;
+
+// If on Pantheon...
+if ( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ) {
+	add_filter( 'site_status_tests', __NAMESPACE__ . '\\site_health_mods' );
+	add_filter( 'site_status_tests', __NAMESPACE__ . '\\object_cache_tests' );
+}
+
+/**
+ * Modify the Site Health tests.
+ *
+ * @param array $tests The Site Health tests.
+ * @return array
+ */
+function site_health_mods( $tests ) {
+	// Remove checks that aren't relevant to Pantheon environments.
+	unset( $tests['direct']['update_temp_backup_writable'] );
+	unset( $tests['direct']['available_updates_disk_space'] );
+	unset( $tests['async']['background_updates'] );
+	return $tests;
+}
+
+/**
+ * Add object cache tests.
+ *
+ * @param array $tests The Site Health tests.
+ * @return array
+ */
+function object_cache_tests( $tests ) {
+	$tests['direct']['object_cache'] = [
+		'label' => __( 'Object Cache', 'pantheon' ),
+		'test'  => 'test_object_cache',
+	];
+
+	return $tests;
+}
+
+/**
+ * Check for object cache and object cache plugins.
+ *
+ * @return array
+ */
+function test_object_cache() {
+	if ( ! isset( $_ENV['CACHE_HOST'] ) ) {
+		$result = [
+			'label' => __( 'Redis Object Cache', 'pantheon' ),
+			'status' => 'critical',
+			'badge' => [
+				'label' => __( 'Performance', 'pantheon' ),
+				'color' => 'red',
+			],
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( 'Redis object cache is not active for your site.', 'pantheon' )
+			),
+			'test' => 'object_cache',
+		];
+
+		return $result;
+	}
+
+	$wp_redis_active = is_plugin_active( 'wp-redis/wp-redis.php' );
+	$ocp_active = is_plugin_active( 'object-cache-pro/object-cache-pro.php' );
+
+	if ( $wp_redis_active ) {
```

Comment:
suggestion: drop else statements and return result earlier each case. result object could be DRYed up a bit but I care less about that.